src/Core/Config/Util/ConfigFileManager.php

```diff
@@ -229,6 +245,11 @@ public function saveData(Cache $configCache)
 	{
 		$filename = $this->configDir . '/' . self::CONFIG_DATA_FILE;
 
+		// fail at a early stage, if we already know that we cannot save the data
+		if (!$this->dataIsWritable()) {
+			throw new ConfigFileException(sprintf('Cannot open file "%s" in mode c+', $filename));
+		}
+
```

Comment:
Since this is independent from the statement on line 246, I believe it should be moved before.